### PR TITLE
Feature/add code formatter

### PR DESCRIPTION
**Pull request by SofianeBenziane**
## What does this PR do?

Add a code formatter (prettier) that executes at each commit or via make commands

## Tasks

- [x] Add prettier in the package.json
- [x] Create make commands that use prettier
- [x] Create a pre-commit hook that executes prettier

### CODE REVIEW

File: .githooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+FILES=$(git diff --cached --name-only --diff-filter=ACM "*.js" "*.jsx" | sed 's| |\\ |g')
+[ -z "$FILES" ] && exit 0
+
+# Prettify all selected files and store the output
+CHECK=$(./node_modules/.bin/prettier --check --use-tabs --end-of-line crlf "$FILES")
```
Comment (so-mouga) [inline .githooks/pre-commit:6]: Why not run the command make app-cs-check  ?

File: Makefile [hunk 2]
```diff
@@ -102,3 +104,12 @@ app-db-unmigrate:
 ## to show logs from containers, specify "c=service_name" to filter logs by container
 infra-show-logs:
 	docker-compose logs -ft ${c}
+
+## to show files that need to be fixed
+app-cs-check: 
+	$(docker_exec_web) sh -c './node_modules/prettier/bin-prettier.js --check "**/*.js" --use-tabs --end-of-line crlf --ignore-path ./node_modules/**/*.* ./.githooks/ ./bin/ ./config/ ./docs/'
```
Comment (so-mouga) [inline Makefile]: maybe set the config exclude in a config prettier ?

File: .githooks/pre-commit (hunk 1 above)
Comment (SofianeBenziane) [reply .githooks/pre-commit:6]: app-cs-check checks the entire application while this command checks only the added files on git

File: Makefile (hunk 2 above)
Comment (Sbenziane) [reply Makefile]: You're right, that's why I just did it.

Comment (Sbenziane) [reply Makefile]: You're right, that's why I just did it.

Comment (SofianeBenziane) [reply Makefile]: You're right, that's why I just did it.

**Pull request merged**